deploy/tests/test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+KUBE_GLUSTER_DEAMONSET=../kube-templates/glusterfs-daemonset.yaml 
+KUBE_HEKETI_DEPLOYMENT=../kube-templates/heketi-deployment.yaml
+KUBE_HEKETI_SA=../kube-templates/heketi-service-account.yaml 
+OCP_GLUSTER_TEMPLATE=../ocp-templates/glusterfs-template.yaml
+OCP_HEKETI_TEMPLATE=../ocp-templates/heketi-template.yaml
+OCP_HEKETI_SA=../ocp-templates/heketi-service-account.yaml
+TOPOLOGY=../topology.json.sample
+
+# syntax checking can not be done if file has more than one yaml context 
+# so split and call check
+# check kube-templates/heketi-deployment.yaml to understand what I am talking about
+split_yaml_check() {
+  file="${1}"
+  awk '{print $0 > "file" NR}' RS='---'  ${file}
+  count=$(ls -l file* | wc -l)
+  for i in `seq 1 ${count}`
+  do
+    sed -i '1s/^/\-\-\-/' file${i}
+    check "file${i}"
+  done
+}
+
+# yaml syntax check
+check() {
+  file="${1}"
+  TEST=$(python -c 'import yaml,sys;yaml.safe_load(sys.stdin)' < ${file} 2>/dev/null)
+  if [[ $? -ne 0 ]]; then
+    echo "Failed ${file}"
+  fi
+}
+
+# copies the source and adds the mock source. This will be used for testing
+cp ../gk-deploy gk-deploy-unit
+sed -i '16s/^/\nsource mock.sh/' gk-deploy-unit
```

Comment:
I tested it, see https://github.com/obnoxxx/gluster-kubernetes/tree/pr-184-plus . (This also contains a few cleanups in the formatting etc).

But there seem to be issues. the sourcing in the test script does not work immediately.

I noticed a problem: When you just comment out the creation of the patched script, the test runs a little longer but succeeds nonetheless even though no kubectl or oc commands are available.. :-o This needs to be fixed.